packages/grafana-ui/src/types/completion.ts

```diff
@@ -11,10 +13,7 @@ export interface CompletionItemGroup {
    */
   items: CompletionItem[];
 
-  /**
-   * If true, match only by prefix (and not mid-word).
-   */
-  prefixMatch?: boolean;
+  searchFunction?: SearchFunction;
```

Comment:
The `prefixMatch` is external API that can be used by other plugins. Therefore change from prefixMatch -> searchFunction might break plugins (e.g. this is plugin using this API, but not specifically prefixMatch https://github.com/marcusolsson/grafana-json-datasource/blob/main/src/suggestions.ts). So we should probably deprecate `prefixMatch` by adding `* @deprecated use searchFunction instead` and check if `prefixMatch === true` and `searchFunction === undefined` and set `searchFunction = prefixMatch`. Orif we want to do breaking change, then probably Grafana 8 is a good time.

Reply:
Oh this is very good point! It probably makes sense to keep it backwards compatible for some time. It also makes me thinking about the API. Should we expose `searchFunction` allowing to add custom search functions or just an enum to limit the choice and keep search functions internally?

Reply:
I think that enum works perfectly.

Reply:
Converted to enums, only `SearchFunctionType` is exposed now.